src/internal/packager/helm/post-render.go

```diff
@@ -54,8 +54,17 @@ func (h *Helm) newRenderer(ctx context.Context) (*renderer, error) {
 	if kerrors.IsNotFound(err) {
 		rend.namespaces[h.chart.Namespace] = cluster.NewZarfManagedNamespace(h.chart.Namespace)
 	} else if h.cfg.DeployOpts.AdoptExistingResources {
-		namespace.Labels = cluster.AdoptZarfManagedLabels(namespace.Labels)
-		rend.namespaces[h.chart.Namespace] = namespace
+		b, err := json.Marshal(namespace)
```

Comment:
This feels a bit strange, but it seems to be the best way to get then apply a resource with ssa, since the get is a different type

The library has `extractresource` functions, however these will only get resources from the managed fields. Because we're just switching to server side apply now, existing resources will not have managed fields. Additionally, in the case of adoptexistingresources, we can't assume we previously owned the field

https://kubernetes.io/blog/2021/08/06/server-side-apply-ga/#using-server-side-apply-in-a-controller

Reply:
Ended up avoiding the get->apply flows